core/src/sql/part.rs

```diff
@@ -130,12 +188,164 @@ impl fmt::Display for Part {
 				}
 			}
 			Part::Optional => write!(f, "?"),
+			Part::Recurse(v, nest) => {
+				write!(f, ".{{{v}}}")?;
+				if let Some(nest) = nest {
+					write!(f, "({nest})")?;
+				}
+
+				Ok(())
+			}
+			Part::Doc => write!(f, "@"),
+			Part::RepeatRecurse => write!(f, ".@"),
 		}
 	}
 }
 
 // ------------------------------
 
+#[derive(Clone, Debug)]
+pub enum RecursionPlan {
+	Repeat,
+	Destructure(
+		// The destructure parts
+		Vec<DestructurePart>,
+		// Which field contains the repeat symbol
+		Ident,
+		// Path before the repeat symbol
+		Vec<Part>,
+		// The recursion plan
+		Box<RecursionPlan>,
+		// Path after the repeat symbol
+		Vec<Part>,
+	),
```

Comment:
Nitpick: Instead of a tuple, it is better for code readability to use fields in this case. So maybe something like below?
```suggestion
	Destructure{
		// The destructure parts
		destructure: Vec<DestructurePart>,
		// Which field contains the repeat symbol
		repeating_field: Ident,
		// Path before the repeat symbol
		prefix: Vec<Part>,
		// The recursion plan
		recursion_plan: Box<RecursionPlan>,
		// Path after the repeat symbol
		suffix: Vec<Part>,
	},
```

Reply:
Why did I forget this was possible haha, thanks!

Reply:
Fixed in [ad8da37](https://github.com/surrealdb/surrealdb/pull/5086/commits/ad8da375c037d44fdb15b17497d8251297bea418)